controllers/flowlogspipeline/flp_objects.go

```diff
@@ -255,10 +258,70 @@ func (b *builder) addTransformStages(lastStage *config.PipelineBuilderStage) {
 	agg := enrichedStage.Aggregate("aggregate", []api.AggregateDefinition{})
 	agg.EncodePrometheus("prometheus", api.PromEncode{
 		Port:   int(b.desired.PrometheusPort),
-		Prefix: "flp_",
+		Prefix: "netobserv_",
 	})
 }
 
+func (b *builder) mergeMetricsConfiguration(pipeline config.PipelineBuilderStage) {
```

Comment:
Hey @KalmanMeth , @eranra ,
I am sorry to say, I am not really convinced with the approach here :(
I find it hard to follow, and error prone, because we are mixing two different configs, each one being created in a totally different way. I think it creates a problem of consistency that makes the whole harder to reason about. For instance, when looking at how Prometheus, or Loki, or whatever else is configured, there's now several places that seem to indicate a configuration - but only one is trustworthy. 

Also, as pointed below by Olivier, risk of getting duplicate or conflicting configs. The kubernetes enrichment is now performed twice, and both add up. We would need to be very careful, in  future changes, that whatever we want to add or change is done in the right place.

Finally, I did a quick try to see how it would like if the metrics config was created from code. Maybe that's a matter of taste, but I don't find anything shocking there. That's how it would look like:

```go
func (b *builder) addMetrics(lastStage *config.PipelineBuilderStage) {
	// prometheus stage (encode) configuration
	agg := lastStage.Aggregate("aggregate", []api.AggregateDefinition{{
		Name:      "bandwidth_network_service_namespace",
		By:        []string{"service", "SrcK8S_Namespace"},
		Operation: "sum",
		RecordKey: "Bytes",
	}, {
		Name:      "bandwidth_source_subnet",
		By:        []string{"srcSubnet"},
		Operation: "sum",
		RecordKey: "Bytes",
	}, {
		Name:      "dest_service_count",
		By:        []string{"service"},
		Operation: "count",
	}})
	agg.EncodePrometheus("prometheus", api.PromEncode{
		Port:   int(b.desired.PrometheusPort),
		Prefix: "netobserv_",
		Metrics: api.PromMetricsItems{{
			Name:     "bandwidth_per_network_service_per_namespace",
			Type:     "counter",
			Filter:   api.PromMetricsFilter{Key: "name", Value: "bandwidth_network_service_namespace"},
			ValueKey: "recent_op_value",
			Labels:   []string{"by", "aggregate"},
			Buckets:  []float64{},
		}, {
			Name:     "bandwidth_per_source_subnet",
			Type:     "counter",
			Filter:   api.PromMetricsFilter{Key: "name", Value: "bandwidth_source_subnet"},
			ValueKey: "recent_op_value",
			Labels:   []string{"by", "aggregate"},
			Buckets:  []float64{},
		}, {
			Name:     "network_service_count",
			Type:     "counter",
			Filter:   api.PromMetricsFilter{Key: "name", Value: "dest_service_count"},
			ValueKey: "recent_count",
			Labels:   []string{"by", "aggregate"},
			Buckets:  []float64{},
		}},
	})
}
```

(code not tested)

I think it's pretty concise, has the advantage to put everything in the same place and avoids doing a config merge.
Also, something else that we don't take advantage of at the moment, but could in the future, is if we want to control aspects of the produced metrics from the operator custom resource => it's better to have them created from code then.

Reply:
@jotak @eranra @OlivierCazade I think we need to have a discussion on this.
Certainly the configuration can be hard-coded, but the idea of a config file is to allow flexibility, and to allow changes without having to change code and recompile. With a configuration file, different installations can produce a different set of metrics.

Reply:
As discussed, we placed all the enrichment in the Operator and do not merge enrichments. The fields for extract_aggregate and encode_prom are taken completely from the metrics configuration. The call to confgenerator to create the metrics configuration is now incorporated into the Operator.